src/dirbrowserformaction.cpp

```diff
@@ -335,14 +335,14 @@ void DirBrowserFormAction::add_directory(
 	}
 }
 
-std::string DirBrowserFormAction::get_formatted_dirname(std::string dirname,
+Filepath DirBrowserFormAction::get_formatted_dirname(const Filepath& dirname,
 	mode_t mode)
 {
 	const auto suffix = file_system::mode_suffix(mode);
 	if (suffix.has_value()) {
 		return strprintf::fmt("%s%c", dirname, suffix.value());
```

Comment:
Please add https://doc.rust-lang.org/nightly/std/path/struct.PathBuf.html#method.set_extension to `Filepath` and rewrite this line to use that method instead.

Reply:
https://github.com/newsboat/newsboat/pull/2561/commits/518a6a395021e6d62051f86352b6db6d9bf6f35c

Reply:
Thank you for adding the test as well!